File: samples/python/08objects/objects.py
import math

class Shape:
    def __init__(self):
        pass

    def area(self):
        pass

    def perimeter(self):
        pass


# For the purposes of this:
#  _______________
#  |              | side A
#  |______________|
#         side B
class Rectangle(Shape):
    side_a = 0
    side_b = 0

    def __init__(self, side_a, side_b):
        self.side_a = side_a
        self.side_b = side_b

    def area(self):
        return self.side_a * self.side_b

    def perimeter(self):
        return self.side_a * 2 + self.side_b * 2

class Sqaure(Rectangle):
    def __init__(self, side):
        super().__init__(side, side)

#  A Rigth triangle as follows
#
#   |\
# A | \
#   |__\
#     B
class Triangle(Shape):
    side_a = 0
    side_b = 0
    side_c = 0

    def __init__(self, side_a, side_b):
        self.side_a = side_a
        self.side_b = side_b
        self.side_c = math.sqrt(side_a ** 2 + side_b ** 2)

    def area(self):
        return (0.5) * self.side_a * self.side_b

    def perimeter(self):
        return self.side_a + self.side_b + self.side_c

#  For The  purposes fo this:
#   ______________
#  |              |\
#  |______________|_\
#  (they follow the same shape described for rectangle and triangle)
class Trapezoid(Shape):
    rectangle_part = None
    triangle_part = None
    def __init__(self, rectangle_part, triangle_part):
        if rectangle_part.side_a != triangle_part.side_a:
            raise Exception('Rectangle and Triangle Sides do not match up')
        self.rectangle_part = rectangle_part
        self.triangle_part = triangle_part
    
    def area(self):
        return self.rectangle_part.area() + \
            self.triangle_part.area()

    def perimeter(self):
        return self.rectangle_part.perimeter() - self.rectangle_part.side_a + \
            self.triangle_part.perimeter() - self.triangle_part.side_a


r = Rectangle(2, 5)

s = Sqaure(2)

t = Triangle(2, 2)

tr = Trapezoid(r, t)

print(r.area(), r.perimeter())

print(s.area(), s.perimeter())

print(t.area(), t.perimeter())

print(tr.area(), tr.perimeter())

File: samples/js/08objects/objects.js
'use strict';

class Shape {
    constructor() {
    }
    area() {}
    perimeter() {}
}

class Circle extends Shape {
    radius = 0;

    constructor(radius) {
        super();
        this.radius = radius;
    }

    area() {
        return Math.PI * this.radius * this.radius;
    }

    perimeter() {
        return 2 * Math.PI * this.radius;
    }
}

class Rectangle extends Shape {
    sideA = 0;
    sideB = 0;

    constructor(sideA, sideB) {
        super()
        this.sideA = sideA;
        this.sideB = sideB;
    }

    area() {
        return this.sideA * this.sideB;
    }

    perimeter() {
        return this.sideA * 2 + this.sideB * 2;
    }
}

class Square extends Rectangle {
    constructor(length) {
        super(length, length);
    }
}

class Triangle extends Shape {
    sideA = 0;
    sideB = 0;
    sideC = 0;

    constructor(sideA, sideB) {
        super();
        this.sideA = sideA;
        this.sideB = sideB;
        this.sideC = Math.sqrt(this.sideA*this.sideA + this.sideB * this.sideB);
    }

    area() {
        return this.sideA * this.sideB * 0.5;
    }

    perimeter() {
        return this.sideA + this.sideB + this.sideC;
    }
}

class Trapezoid extends Shape {
    rectanglePart = null;
    trianglePart = null;

    constructor(rectanglePart, trianglePart) {
        if (rectanglePart.sideA != trianglePart.sideA) {
            throw Error('Rectangle And Triangle must have the same length sideA');
        }
        super();
        this.rectanglePart = rectanglePart;
        this.trianglePart = trianglePart;
    }

    area() {
        return this.rectanglePart.area() +
            this.trianglePart.area();
    }

    perimeter() {
        return this.rectanglePart.perimeter() - this.rectanglePart.sideA +
            this.trianglePart.perimeter() - this.trianglePart.sideA;
    }
}

const r = new Rectangle(2, 5);
const s = new Square(4);
const t = new Triangle(2, 5);
const tz = new Trapezoid(r, t);
const c = new Circle(4);

console.log('Radius', r.area(), r.perimeter());
console.log('Square', s.area(), s.perimeter());
console.log('Triangle', t.area(), t.perimeter());
console.log('Trapezoid', tz.area(), tz.perimeter());
console.log('Circle', c.area(), c.perimeter());